indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
-            p = Paragraph( _("Table of contents"), style1)
-            self._story.append(p)
-            self._story.append(Spacer(inch, 1*cm))
-            if len(self._showSessions)>0:
-                style2=ParagraphStyle({})
-                style2.fontSize=modifiedFontSize(14, self._fontsize)
-                style2.leading=modifiedFontSize(10, self._fontsize)
-                style2.alignment=TA_CENTER
-                caption=[]
-                for sessionId in self._showSessions:
-                    session=self._conf.getSessionById(sessionId)
-                    if session is not None:
-                        caption.append(session.getTitle())
-                p=Paragraph("%s"%"\n".join(caption),style2)
+            self._story.append(Spacer(inch, 1 * cm))
+            self._story.append(Paragraph(_("Table of contents"), style1))
+            self._story.append(Spacer(inch, 1 * cm))
+            if self._showSessions:
+                style2 = ParagraphStyle({})
+                style2.fontSize = modifiedFontSize(14, self._fontsize)
+                style2.leading = modifiedFontSize(10, self._fontsize)
+                style2.alignment = TA_CENTER
+                caption = []
+                for session_id in self._showSessions:
+                    sess = self._conf.as_event.sessions.filter_by(id=session_id, is_deleted=False).first()
+                    if sess:
+                        caption.append(sess.title)
+                p = Paragraph("\n".join(caption), style2)
                 self._story.append(p)
-            self._story.append(Spacer(inch, 2*cm))
+            self._story.append(Spacer(inch, 2 * cm))
             self._story.append(self._toc)
             self._story.append(PageBreak())
 
-    def firstPage(self,c,doc):
+    def firstPage(self, c, doc):
         if self._ttPDFFormat.showCoverPage():
             c.saveState()
             if self._ttPDFFormat.showLogo():
                 self._drawLogo(c, False)
+
             height = self._drawWrappedString(c, self._conf.getTitle())
-            c.setFont('Times-Bold',modifiedFontSize(15, self._fontsize))
-            height-=2*cm
-            c.drawCentredString(self._PAGE_WIDTH/2.0,height,
-                    "%s - %s"%(self._conf.getAdjustedStartDate(self._tz).strftime("%A %d %B %Y"),
-                    self._conf.getAdjustedEndDate(self._tz).strftime("%A %d %B %Y")))
-            if self._conf.getLocation():
-                height-=2*cm
-                c.drawCentredString(self._PAGE_WIDTH/2.0,height,escape(self._conf.getLocation().getName()))
+            c.setFont('Times-Bold', modifiedFontSize(15, self._fontsize))
+            height -= 2 * cm
+            c.drawCentredString(self._PAGE_WIDTH / 2.0, height,
+                                "{} - {}".format(self._conf.getAdjustedStartDate(self._tz).strftime("%A %d %B %Y"),
+                                                 self._conf.getAdjustedEndDate(self._tz).strftime("%A %d %B %Y")))
```

Comment:
We could use `Event.start_dt` and `Event.end_dt` here, like we've done for the meeting pages (we'll have to eventually do it anyway). I'd leave it for the end, though.
I wonder if we should have `get_local_start_dt(tz=None)` and `get_local_end_dt(tz=None)` methods, as we need to convert between timezones very often (also in the templates). Once again, not urgent. /cc @ThiefMaster